src/subscribables/dependentObservable.js

```diff
@@ -250,23 +242,62 @@ ko.computed = ko.dependentObservable = function (evaluatorFunctionOrOptions, eva
         pure = true;
         isSleeping = true;     // Starts off sleeping; will awake on the first subscription
         dependentObservable.beforeSubscriptionAdd = function (event) {
-            // If asleep, wake up the computed and evaluate to register any dependencies.
-            if (event == 'change' && isSleeping) {
+            // If asleep, wake up the computed by subscribing to any dependencies.
+            if (!_isDisposed && isSleeping && event == 'change') {
                 isSleeping = false;
-                _needsEvaluation = true;
-                evaluateImmediate(true /* suppressChangeNotification */);
-                if (!_isDisposed) {     // test since evaluateImmediate could trigger disposal
+                if (_needsEvaluation || haveDependenciesChanged()) {
+                    dependencyTracking = null;
+                    _dependenciesCount = 0;
+                    _needsEvaluation = true;
+                    evaluateImmediate();
+                } else {
+                    // First put the dependencies in order
```

Comment:
In Chrome, at least, when the object keys are numeric, it re-orders them in numeric order. In an earlier iteration of this feature, I had a test that failed without this code but it was mainly testing another feature that I didn't include. I'll see if I can modify the test to show the need for this.